crates/api/src/v1/monitor.rs

```diff
@@ -0,0 +1,236 @@
+//! Types relating to the monitor API.
+
+use crate::Status;
+use serde::{de::Unexpected, Deserialize, Serialize, Serializer};
+use std::borrow::Cow;
+use thiserror::Error;
+use warg_crypto::hash::AnyHash;
+use warg_crypto::signing;
+use warg_protocol::registry::RegistryLen;
+
+/// Represents checkpoint verification response in one of the following states:
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "state", rename_all = "camelCase")]
+#[allow(clippy::large_enum_variant)]
+pub enum CheckpointVerificationResponse {
+    /// The checkpoint is verified.
+    #[serde(rename_all = "camelCase")]
+    Verified,
+    /// The checkpoint is verified but has not verified the signature.
+    #[serde(rename_all = "camelCase")]
+    VerifiedButSignatureUnverified,
+    /// The checkpoint is unverified but should retry verification.
+    #[serde(rename_all = "camelCase")]
+    Retry {
+        /// Instructs to retry but wait the specified number of seconds.
+        wait_seconds: u16,
+    },
+    /// The checkpoint is unverified and should not retry.
+    #[serde(rename_all = "camelCase")]
+    Unverified,
```

Comment:
What does this state indicate? I thought it was "we haven't seen that checkpoint" but it looks like that is covered by `MonitorError::CheckpointNotFound`

Reply:
How would that be different than `CheckpointNotFound`? Anyway if you're planning to change these around I'll wait to see where it ends up

Reply:
Refactored, but I might have misunderstood the original question.